include/tvm/runtime/container.h

```diff
@@ -1554,6 +1593,954 @@ struct PackedFuncValueConverter<Optional<T>> {
+    /*! \brief Compare iterators */
+    bool operator!=(const T& other) const { return !(*this == other); }
+    /*! \brief De-reference iterators */
+    reference operator*() const { return ListNode(i, self).Data(); }
+    /*! \brief Pointer-to of iterators */
+    pointer operator->() const { return &ListNode(i, self).Data(); }
+    /*! \brief Prefix self increment */
+    T& operator++() {
+      if (self == nullptr || i > self->slots_) {
+        return static_cast<T&>(*this);
+      }
+      for (++i; i <= self->slots_; ++i) {
+        if (!ListNode(i, self).IsEmpty()) {
+          return static_cast<T&>(*this);
+        }
+      }
+      return static_cast<T&>(*this);
+    }
+    /*! \brief Prefix self decrement */
+    T& operator--() {
+      if (self == nullptr || i > self->slots_ + 1) {
+        return static_cast<T&>(*this);
+      }
+      while (i-- != 0) {
+        if (!ListNode(i, self).IsEmpty()) {
+          return static_cast<T&>(*this);
+        }
+      }
+      i = self->slots_ + 1;
+      return static_cast<T&>(*this);
+    }
+    /*! \brief Suffix self increment */
+    T operator++(int) {
+      T copy = static_cast<T&>(*this);
+      ++(*this);
+      return copy;
+    }
+    /*! \brief Suffix self decrement */
+    T operator--(int) {
+      T copy = static_cast<T&>(*this);
+      --(*this);
+      return copy;
+    }
+    /*! \brief Constructor */
+    IteratorBase(uint64_t _i, const MapNode* _self) : i(_i), self(_self) {
+      if (self != nullptr) {
+        for (; i <= self->slots_; ++i) {
+          if (!ListNode(i, self).IsEmpty()) {
+            break;
+          }
+        }
+      }
+    }
+    /*! \brief The position on the array */
+    uint64_t i;
+    /*! \brief The container it points to */
+    const MapNode* self;
+  };
+
+  static_assert(sizeof(KVType) == 16 || sizeof(KVType) == 8, "sizeof(KVType) incorrect");
+  static_assert(sizeof(Block) == kBlockCap * (sizeof(KVType) + 1), "sizeof(Block) incorrect");
+  static_assert(std::is_standard_layout<KVType>::value, "KVType is not standard layout");
+  static_assert(std::is_standard_layout<Block>::value, "Block is not standard layout");
+
+ public:
+  /*! \brief The iterator of hash map */
+  class iterator : public IteratorBase<iterator> {
+   public:
+    using iterator_category = std::bidirectional_iterator_tag;
+    using difference_type = int64_t;
+    using value_type = typename KVType::TStl;
+    using pointer = value_type*;
+    using reference = value_type&;
+
+    iterator() = default;
+
+   private:
+    iterator(uint64_t i, const MapNode* self) : IteratorBase(i, self) {}
+    friend class MapNode;
+  };
+
+ protected:
+  /*! \brief array of data blocks */
+  Block* data_;
+  /*! \brief number of slots minus 1 */
+  uint64_t slots_;
+  /*! \brief number of entries in the container */
+  uint64_t size_;
+  /*! \brief fib shift in Fibonacci Hashing */
+  uint8_t fib_;
```

Comment:
if the shift is always in size_t, we should not to store in uint8_t(will results in runtime conversion), consider to store as int32_t instead.

Reply:
fib_shift_